SocketIOClientSwift/SocketIOClient.swift

```diff
@@ -86,24 +68,19 @@ public final class SocketIOClient: NSObject, SocketEngineClient, SocketLogClient
         self.socketURL = socketURL
         self.opts = opts
         
-        // Set options
-        if let sessionDelegate = opts?["sessionDelegate"] as? NSURLSessionDelegate {
-            self.sessionDelegate = sessionDelegate
-        }
-        
         if let connectParams = opts?["connectParams"] as? [String: AnyObject] {
             self.connectParams = connectParams
         }
         
+        if let logger = opts?["logger"] as? SocketLogger {
+            Logger = logger
+        }
+        
         if let log = opts?["log"] as? Bool {
-            self.log = log
+            Logger.log = log
```

Comment:
Seems like all files make use this singleton, but it gets set when creating a `SocketIOClient`? That's probably going to lead to some confusing behavior. I'd recommend not making it a singleton, or at least not making it mutable.